src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,153 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private static final int NEGATIVE_INDEX_MARKER = -1;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.elementData = new Object[initialCapacity];
+        } else if (initialCapacity == 0) {
+            this.elementData = EMPTY_ELEMENTDATA;
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: "
+                    + initialCapacity);
+        }
+    }
+
+    public ArrayList() {
+        this.elementData = DEFAULTCAPACITY_EMPTY_ELEMENTDATA;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < this.elementData.length) {
+            elementData[size] = value;
+            size++;
+        } else {
+            add(value, size);
+        }
```

Comment:
Don't complicate if-else construction. Consider refactoring to avoid code duplication and improve readability. The logic to ensure the array has enough capacity could be encapsulated in a separate method and reused in `add(T value, int index)`.